### ISSUE TITLE
How to train YOLOv8 with single class without considering cls loss

### ISSUE THREAD

**Issue by fuqianya** [ultralytics/ultralytics#13132]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi,

Firstly, thank you for the excellent work on this project. I am currently using YOLOv8 and have a specific requirement where I need to focus solely on the localization of an object in a single-class dataset, ignoring the class identification.

To configure YOLOv8 for localization:

I set the `cls_loss` weight to `0.0` and enabled `single-cls` mode.

```
from ultralytics import YOLO

model = YOLO('yolov8l.pt').load('./yolov8l.pt')
# The VisDrone.yaml has been processed to have one class
results = model.train(data='VisDrone.yaml', epochs=100, imgsz=640, device=[7], cls=0.0, single_cls=True)
```

However, this resulted in a low mAP, as illustrated below:

<img width="1248" alt="image" src="https://github.com/ultralytics/ultralytics/assets/45023083/f6851157-b7ef-493d-bfa1-cdc7dcefe2f6">

When I revert the `cls_loss` weight to the default value of `0.5`, the mAP returns to normal levels:

```
from ultralytics import YOLO

model = YOLO('yolov8l.pt').load('./yolov8l.pt')
# The VisDrone.yaml has been processed to have one class
results = model.train(data='VisDrone.yaml', epochs=100, imgsz=640, device=[7], cls=0.5, single_cls=True)
```
<img width="1212" alt="image" src="https://github.com/ultralytics/ultralytics/assets/45023083/a0ccf7a6-e74f-4f43-aee3-c691e0f842b4">

Given that my task does not require class loss, I believe setting `cls_loss` to `0.0` is appropriate, but the results suggest otherwise. (Maybe the prediction (e.g., val) stage considers the class prediction, but how to turn off it? )

Could you please advise on how best to train a YOLOv8 model for localization tasks under these conditions? Any guidance or suggestions would be greatly appreciated.

Thank you!

**Comment by glenn-jocher** [comment]
Hi there! 😊

Thank you for your kind words and for using YOLOv8. To focus solely on localization and ignore class loss, you can set the `cls` weight to `0.0`, but ensure that the model is correctly configured to handle single-class scenarios.

Here's a refined approach:

```python
from ultralytics import YOLO

# Load the model
model = YOLO('yolov8l.pt').load('./yolov8l.pt')

# Train with cls loss set to 0.0 and single_cls enabled
results = model.train(data='VisDrone.yaml', epochs=100, imgsz=640, device=[7], cls=0.0, single_cls=True)
```

If you're still seeing low mAP, it might be due to the evaluation stage considering class predictions. Ensure your validation configuration aligns with your training setup.

Feel free to tweak and experiment with these settings. If issues persist, the community and Ultralytics team are here to help! 🚀

Happy training!
